src/common/utils/setup-helpers.js

```diff
@@ -175,15 +175,27 @@ function isDebugMode(): boolean {
  * @param {KPOptionsObject} options - kaltura player options
  * @returns {void}
  */
-function setLogLevel(options: KPOptionsObject): void {
+function setLogOptions(options: KPOptionsObject): void {
+  if (!Utils.Object.getPropertyPath(options, 'ui.log')) Utils.Object.createPropertyPath(options, 'ui.log', {});
+  if (!Utils.Object.getPropertyPath(options, 'provider.log')) Utils.Object.createPropertyPath(options, 'provider.log', {});
+  if (!Utils.Object.getPropertyPath(options, 'log')) Utils.Object.createPropertyPath(options, 'log', {});
+
+  if (options.log && typeof options.log.handler === 'function') {
+    setLogHandler(options.log.handler);
+    // $FlowFixMe
+    options.ui.log.handler = options.provider.log.handler = options.log.handler;
+  }
+
   let logLevelObj: LogLevelObject = LogLevel.ERROR;
-  if (isDebugMode()) {
+  if (options.log && isDebugMode()) {
```

Comment:
1. `options.log` must be. you just create it a few lines above 
2. `isDebugMode` got from the url so it must to be applied. with no `options.log` condition

Reply:
does flow require it?

Reply:
Flow require